ingestion/src/metadata/ingestion/source/dashboard/superset/db_source.py

```diff
@@ -0,0 +1,247 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""
+Superset source module
+"""
+
+import json
+import traceback
+from typing import Iterable, List, Optional
+
+from sqlalchemy.engine import Engine
+
+from metadata.generated.schema.api.data.createChart import CreateChartRequest
+from metadata.generated.schema.api.data.createDashboard import CreateDashboardRequest
+from metadata.generated.schema.api.lineage.addLineage import AddLineageRequest
+from metadata.generated.schema.entity.data.chart import ChartType
+from metadata.generated.schema.entity.data.dashboard import (
+    Dashboard as Lineage_Dashboard,
+)
+from metadata.generated.schema.entity.data.table import Table
+from metadata.generated.schema.entity.services.connections.dashboard.supersetConnection import (
+    SupersetConnection,
+)
+from metadata.generated.schema.entity.services.connections.metadata.openMetadataConnection import (
+    OpenMetadataConnection,
+)
+from metadata.generated.schema.entity.services.dashboardService import (
+    DashboardServiceType,
+)
+from metadata.generated.schema.metadataIngestion.workflow import (
+    Source as WorkflowSource,
+)
+from metadata.generated.schema.type.entityReference import EntityReference
+from metadata.ingestion.api.source import InvalidSourceException, SourceStatus
+from metadata.ingestion.source.dashboard.dashboard_service import DashboardServiceSource
+from metadata.ingestion.source.dashboard.superset.queries import (
+    FETCH_ALL_CHARTS,
+    FETCH_DASHBOARDS,
+)
+from metadata.utils import fqn
+from metadata.utils.helpers import get_standard_chart_type
+from metadata.utils.logger import ingestion_logger
+
+logger = ingestion_logger()
+
+
+class SupersetDBSource(DashboardServiceSource):
+    """
+    Superset DB Source Class
+    """
+
+    config: WorkflowSource
+    metadata_config: OpenMetadataConnection
+    status: SourceStatus
+    platform = "superset"
+    service_type = DashboardServiceType.Superset.value
+
+    def __init__(self, config: WorkflowSource, metadata_config: OpenMetadataConnection):
+        super().__init__(config, metadata_config)
+        self.all_charts = {}
+        self.service_connection: SupersetConnection = self.service_connection
```

Comment:
Sonar cloud bug here, 😅 might be that it's validating the schema here, but we already have that mechanismon L76-79

Reply:
we can remove this i guess